### PR TITLE
[CM] Better readability and maintainability: rename variables, move code to more logical places 🔧

### PR DESCRIPTION
**Pull request by destogl**
- Move 'manage_switch' method and 'do_switch' flag to more appropriate positions.
- Make variable name clearer: rename 'request' to 'controller_name'.

### CODE REVIEW

File: controller_manager/src/controller_manager.cpp [hunk 1]
```diff
@@ -1567,16 +1538,14 @@ void ControllerManager::activate_controllers(
...
+      resource_manager_->make_controller_reference_interfaces_available(controller_name);
     }
 
     if (controller->is_async())
     {
       async_controller_threads_.at(controller_name)->activate();
     }
   }
-  // All controllers activated, switching done
-  switch_params_.do_switch = false;
```
Comment (bmagyar) [inline controller_manager/src/controller_manager.cpp:1579]: this changes behaviour, nicht wahr?

I see it was moved into `manage_switch()` to the end

**Review by bmagyar — CHANGES_REQUESTED** [review]

File: controller_manager/src/controller_manager.cpp (hunk 1 above)
Comment (destogl) [reply controller_manager/src/controller_manager.cpp:1579]: No, it should not change behavior, but just put it in a more logical place.

Comment (saikishor) [reply controller_manager/src/controller_manager.cpp:1579]: I think what you said makes sense. It should be at the end of the `manage_switch` method.

**Review by saikishor — APPROVED** [review]
LGTM

**Review by olivier-stasse — APPROVED** [review]
LGTM

**Comment by olivier-stasse** [comment]
Actually the CI is failing.

File: controller_manager/src/controller_manager.cpp [hunk 2]
```diff
@@ -1454,17 +1423,17 @@ void ControllerManager::activate_controllers(
   const std::vector<ControllerSpec> & rt_controller_list,
   const std::vector<std::string> controllers_to_activate)
 {
-  for (const auto & request : controllers_to_activate)
+  for (const auto & controller_name : controllers_to_activate)
```
Comment (olivier-stasse) [inline controller_manager/src/controller_manager.cpp:1426]: You have a problem here as '''controller_name''' is later defined in line 1440.

**Review by olivier-stasse — CHANGES_REQUESTED** [review]
You should remove the auto declaration in 1440.
The code does not compile in the CI for a good reason.

File: controller_manager/src/controller_manager.cpp (hunk 2 above)
Comment (saikishor) [reply controller_manager/src/controller_manager.cpp:1426]: I think this is the new shadowing errors that we have enabled

Comment (destogl) [reply controller_manager/src/controller_manager.cpp:1426]: @bmagyar sabotaged the PR since he merged uncarefully master into PR.

Comment (bmagyar) [reply controller_manager/src/controller_manager.cpp:1430]: I'm planning to take a new online course "Maintainership for beginners" very soon! :D

**Review by bmagyar — APPROVED** [review]
thanks!

**Pull request merged**